### PR TITLE
Disabled upload optimization due to bad performance

### PR DESCRIPTION
**Pull request by emilwestergren**

### CODE REVIEW

File: source/draw/gpu/opengl/AndroidContext.ooc [hunk 1]
```diff
@@ -38,7 +38,8 @@ AndroidContext: class extends OpenGLContext {
 	}
 	createGpuImage: override func (rasterImage: RasterImage) -> GpuImage {
 		result: GpuImage
-		if (rasterImage instanceOf?(GraphicBufferYuv420Semiplanar)) {
+		//This optimization is currently disabled since it's only faster for 4k resolution
+		if (false && rasterImage instanceOf?(GraphicBufferYuv420Semiplanar)) {
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Can we do something less confusing than `if (false` here? How about a version flag or a setting, somewhere?

Comment (emilwestergren) [inline source/draw/gpu/opengl/AndroidContext.ooc]: That's why I added the comment. I could add a version flag around the entire function instead I guess.

Comment (davidhesselbom) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Would making a decision based on the size of the image, right here in the function, be a bad idea?

Comment (emilwestergren) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Yes, I thought of that too, but the decision will be very dependent of the platform so it's probably a bad idea.

Comment (davidhesselbom) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Alright, then make it a version flag. Not sure what to name the flag, though...

Comment (emilwestergren) [inline source/draw/gpu/opengl/AndroidContext.ooc]: It seems you can't have a whole function inside a version block though.

Comment (davidhesselbom) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Correct... but you can version parts of it.

**Comment by emilwestergren** [comment]
Updated.

File: source/draw/gpu/opengl/AndroidContext.ooc [hunk 2]
```diff
@@ -38,14 +38,19 @@ AndroidContext: class extends OpenGLContext {
 	}
 	createGpuImage: override func (rasterImage: RasterImage) -> GpuImage {
 		result: GpuImage
-		if (rasterImage instanceOf?(GraphicBufferYuv420Semiplanar)) {
-			graphicBufferImage := rasterImage as GraphicBufferYuv420Semiplanar
-			rgba := graphicBufferImage toRgba(this)
-			result = this unpackBgraToYuv420Semiplanar(rgba, rasterImage size)
-			rgba free()
+		version (graphicbufferupload) {
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/AndroidContext.ooc]: @jonathanudd , what do you think of the name of this flag?

Comment (jonathanudd) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Debug flags should be named with the prefix `debug` and then what's it's debugging. In this case something like `debugGraphicbufferupload` or similar.

Comment (emilwestergren) [inline source/draw/gpu/opengl/AndroidContext.ooc]: But this is not really a debugflag, more of a settings/feature flag.

Comment (davidhesselbom) [inline source/draw/gpu/opengl/AndroidContext.ooc]: Right, but it's not really a feature flag, is it?

Comment (emilwestergren) [inline source/draw/gpu/opengl/AndroidContext.ooc]: No, it's probably best described as an optimization flag.

**Comment by emilwestergren** [comment]
Updated. We decided to use `opti` as prefix to optimization flags.

**Pull request merged**